datafusion/core/tests/sql/functions.rs

```diff
@@ -124,43 +124,110 @@ async fn query_array() -> Result<()> {
             Arc::new(StringArray::from_slice(&["", "a", "aa", "aaa"])),
             Arc::new(Int32Array::from(vec![Some(0), Some(1), None, Some(3)])),
         ],
-    )?;
+    )
+    .unwrap();
 
-    let table = MemTable::try_new(schema, vec![vec![data]])?;
+    let table = MemTable::try_new(schema, vec![vec![data]]).unwrap();
 
     let ctx = SessionContext::new();
-    ctx.register_table("test", Arc::new(table))?;
-    let sql = "SELECT array(c1, cast(c2 as varchar)) FROM test";
+    ctx.register_table("test", Arc::new(table)).unwrap();
+    ctx
+}
+
+#[tokio::test]
+async fn query_array() {
+    let ctx = array_context();
+    let sql = "SELECT array[c1, cast(c2 as varchar)] FROM test";
```

Comment:
Prior to this PR, this query would error (only constants were supported in `array[]` syntax. cc @ovr

Reply:
👍